### PR TITLE
[Bugfix] Fix zigzag logits PCC comparison in test_prefill_transformer

### PR DESCRIPTION
**Pull request by ddjekicTT**
### Summary
when `is_balanced = True`, `reverse_reorder_tensor_chunks(tensor, chunk_order, seq_dim=-2)` undoes the zigzag shuffle so that the host-side outputs are back in the original token order before comparing against the reference trace. 

The bug was that the loop applied this same reorder to **every** tensor, including logits. But logits has dim -2 of size 1, so:

`chunk_size = seq_len // num_chunks = 1 // 8 = 0`

With `chunk_size = 0`, every chunk slice is empty `(tensor[..., 0:0, :])`, and concatenating 8 empty slices yields shape `[1, 0, vocab]` — the logits are gone.

### CI Status

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-deepseek-prefill-tests.yaml?query=branch:ddjekic/test_prefill_transformer_logits_fix)

### CODE REVIEW

**Review by ipotkonjak-tt — APPROVED** [review]

File: models/demos/deepseek_v3_d_p/tt/tt_prefill_transformer.py [hunk 1]
```diff
@@ -295,9 +295,13 @@ def forward(
             intermediates["lm_head"] = logits_host
             intermediates["logits"] = first_token_logits
 
-        # Reorder intermediates if balanced
+        # Reorder intermediates if balanced. Skip reordering for logits and lm_head in zigzag mode.
+        no_reorder_keys = {"logits", "lm_head"}
         if return_intermediates and self.is_balanced:
             for key, tensor in intermediates.items():
+                if key in no_reorder_keys:
```
[the remark this answers is not in the document]
Comment (ddjekicTT) [reply models/demos/deepseek_v3_d_p/tt/tt_prefill_transformer.py:302]: not a case we are interested in for prefill

**Review by pavlepopovic — APPROVED** [review]

**Review by dpopovTT — APPROVED** [review]
